### PR TITLE
Feat(PPDSC-2315): implement Applitools spike recommendations

### PR DESCRIPTION
**Pull request by mstuartf**
[PPDSC-2315](https://nidigitalsolutions.jira.com/browse/PPDSC-2315)

**What**

1. Background - implementation of Applitools spike recommendations
2. What did you do
- Add comments in code / docs explaining the importance of the Automated batch closing feature of the Applitools GitHub integration.
- Simplify Applitools config (shared config in env variables, suite-specific config in config files, no config in package.json scripts).
- Add component and docs site tests to the same batch.
- Add custom property `suite` to distinguish comp and doc tests.
4. What does the reviewers should expect
- Comp and doc site tests should be in the same batch with `suite` prop to distinguish them. 
- Should not need to re-run job after resolving a test failure. 

**I have done:**
- [ ] Written unit tests against changes
- [ ] Written functional tests against the component and/or NewsKit site
- [ ] Updated relevant documentation

**I have tested manually:**
- [ ] The feature's functionality is working as expected on Chrome, Firefox, Safari and Edge
- [ ] The screen reader reads and flows through the elements as expected.
- [ ] There are no new errors in the browser console coming from this PR.
- [ ] When visual test is not added, it renders correctly on different browsers and mobile viewports (Safari, Firefox, small mobile viewport, tablet)
- [ ] The Playground feature is working as expected

**Before:**

**After:**

**Who should review this PR:**

**How to test:**

### CODE REVIEW

**Comment by newskit-engineering** [comment]
You can preview these changes on:
  Documentation Site: http://ncu-newskit-docs.s3-website-eu-west-1.amazonaws.com/ppdsc-2315-applitools-config
  Storybook: http://ncu-newskit-docs.s3-website-eu-west-1.amazonaws.com/ppdsc-2315-applitools-config/storybook

File: .circleci/config.yml [hunk 1]
```diff
@@ -291,11 +304,7 @@ commands:
           <<: *applitools_defaults
       - run:
           name: Run Applitools Docs Site Visual Tests
-          command: make e2e_visual_test_docs
-      - run:
-          name: Close the Applitools Batch
-          command: curl --request DELETE --dump-header - "https://eyesapi.applitools.com/api/sessions/batches/$APPLITOOLS_BATCH_ID/close/bypointerid/?apiKey=$APPLITOOLS_API_KEY"
-          when: always
+          command: make e2e_visual_test_docs || true
```
Comment (Vanals) [inline .circleci/config.yml:307]: Can we add a comment here regarding why we have `|| true` etc? as is an uncommon bit worth describing IMO. This is what makes it pass even if the tests fail, right?

Comment (mstuartf) [reply .circleci/config.yml:307]: Yep, good point.

**Comment by Vanals** [comment]
To optimize the money saved I guess we are not merging the two jobs in one? for comp and docs?

**Comment by mstuartf** [comment]
> To optimize the money saved I guess we are not merging the two jobs in one? for comp and docs?

Yeah I think it still makes sense to keep them separate. If you only make changes to the doc site there's no need to run component tests.

**Comment by Vanals** [comment]
> > To optimize the money saved I guess we are not merging the two jobs in one? for comp and docs?
> 
> Yeah I think it still makes sense to keep them separate. If you only make changes to the doc site there's no need to run component tests.

Yep, would not make sense for that one as is just one test. I would when you have to run two proably

**Review by Vanals — APPROVED** [review]

**Review by mutebg — APPROVED** [review]

**Pull request merged**